src/shogun/lib/memory.h

```diff
@@ -98,9 +98,9 @@ template <class T> void sg_generic_free(T* ptr)
 }
 
 template <class InputIt, class OutputIt>
-SG_FORCED_INLINE void* memcpy(InputIt dest, OutputIt src, size_t count)
+SG_FORCED_INLINE void* sg_memcpy(InputIt dest, OutputIt src, size_t count)
 {
-	return std::memcpy(static_cast<void*>(dest), static_cast<const void*>(src), count);
+	return shogun::memcpy(static_cast<void*>(dest), static_cast<const void*>(src), count);
```

Comment:
Hey there. I'm afraid I didn't make the task clear enough :)

* We want to rename this `memcpy` to `sg_memcpy` (which you did) that in turn calls `std::memcpy` just as before (i.e., don't make it call `shogun::memcpy` inside - that won't exist anymore once you rename it)
* We want all existing `memcpy` uses in Shogun to use this newly added `sg_memcpy` instead.

See what I mean?